addons/pos_self_order/controllers/orders.py

```diff
@@ -85,88 +81,87 @@ def get_order_taxes(self, order, access_token):
 
     @http.route('/pos-self-order/get-orders', auth='public', type='json', website=True)
-    def get_orders_by_access_token(self, order_access_tokens):
-        orders_sudo = request.env["pos.order"].sudo().search([
+    def get_orders_by_access_token(self, access_token, order_access_tokens):
+        pos_config = self._verify_pos_config(access_token)
+        session = pos_config.current_session_id
+        orders = session.order_ids.filtered_domain([
             ("access_token", "in", order_access_tokens),
             ("date_order", ">=", fields.Datetime.now() - timedelta(days=7)),
         ])
 
-        if not orders_sudo:
+        if not orders:
             raise NotFound("Orders not found")
 
         orders = []
-        for order in orders_sudo:
+        for order in orders:
             orders.append(order._export_for_self_order())
 
         return orders
 
     @http.route('/pos-self-order/get-tables', auth='public', type='json', website=True)
     def get_tables(self, access_token):
-        pos_config_sudo = request.env['pos.config'].sudo().search([('access_token', '=', access_token)], limit=1)
-
-        if not pos_config_sudo or not pos_config_sudo.self_order_table_mode or not pos_config_sudo.has_active_session:
-            raise Unauthorized("Invalid access token")
-
-        tables = pos_config_sudo.floor_ids.table_ids.filtered(lambda t: t.active).read(['id', 'name', 'identifier', 'floor_id'])
+        pos_config = self._verify_pos_config(access_token)
+        tables = pos_config.floor_ids.table_ids.filtered(lambda t: t.active).read(['id', 'name', 'identifier', 'floor_id'])
 
         for table in tables:
             table['floor_name'] = table.get('floor_id')[1]
 
         return tables
 
-    def _process_lines(self, lines, pos_config_sudo, pos_order_id):
+    def _process_lines(self, lines, pos_config, pos_order_id):
         newLines = []
-        pricelist = request.env['product.pricelist'].sudo().browse(pos_config_sudo.pricelist_id.id)
+        pricelist = pos_config.pricelist_id
 
         for line in lines:
-            product_sudo = request.env["product.product"].sudo().browse(int(line.get("product_id")))
+            product = pos_config.env['product.product'].browse(line.get('product_id'))
             # todo take into account the price extra
-            price_unit = pricelist._get_product_price(product_sudo, quantity=1) if pricelist else product_sudo.lst_price
+            price_unit = pricelist._get_product_price(product, quantity=line.get('qty')) if pricelist else product.lst_price
 
-            config_fiscal_pos = pos_config_sudo.default_fiscal_position_id
-            selected_account_tax = config_fiscal_pos.map_tax(product_sudo.taxes_id) if config_fiscal_pos else product_sudo.taxes_id
+            config_fiscal_pos = pos_config.default_fiscal_position_id
+            selected_account_tax = config_fiscal_pos.map_tax(product.taxes_id) if config_fiscal_pos else product.taxes_id
```

Comment:
It may happen that `config_fiscal_pos` and `product.taxes_id` are not set. 
Could this be a problem?

Reply:
It's fine. When the fiscal_pos is empty, the product taxes are used directly (empty or not).